src/trs/lexicon.rs

```diff
@@ -1,162 +1,182 @@
-///! Hindley-Milner Typing for First-Order Term Rewriting Systems (no abstraction)
-///!
-///! Much thanks to:
-///! - https://github.com/rob-smallshire/hindley-milner-python
-///! - https://en.wikipedia.org/wiki/Hindley%E2%80%93Milner_type_system
-///! - (TAPL; Pierce, 2002, ch. 22)
 use itertools::Itertools;
-use polytype::{self, Context, Type, TypeSchema, Variable as TypeVar};
+use polytype::{Context, Type, TypeSchema, Variable as TypeVar};
 use rand::{thread_rng, Rng};
 use std::f64::NEG_INFINITY;
 use std::fmt;
 use std::iter::{once, repeat};
-use term_rewriting::{Atom, MergeStrategy, Operator, Rule, Signature, Term, Variable, TRS as UntypedTRS};
+use std::sync::{Arc, RwLock};
+use term_rewriting::{Atom, Operator, Rule, Signature, Term, Variable, TRS};
```

Comment:
If we go with the module renaming suggested elsewhere, you'll want to bring back `TRS as UntypedTRS`, which is nicely descriptive.